components/BottomNavigation/examples/MDCBottomNavigationBarController.m

```diff
@@ -0,0 +1,278 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBottomNavigationBarController.h"
+
+@interface MDCBottomNavigationBarController ()
+
+/** The view that hosts the content for the selected view controller **/
+@property(nonatomic, strong) UIView *content;
+
+/** Constrains the navigation bar's height to the constant value of this property **/
+@property(nonatomic, strong) NSLayoutConstraint *navigationBarHeightConstraint;
+
+@end
+
+@implementation MDCBottomNavigationBarController
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _navigationBar = [[MDCBottomNavigationBar alloc] init];
+    _content = [[UIView alloc] init];
+    _viewControllers = @[];
+    _selectedIndex = NSNotFound;
+  }
+
+  return self;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  _navigationBar.delegate = self;
+
+  // Add subviews and create their constraints
+  [self.view addSubview:_navigationBar];
+  [self.view addSubview:_content];
+  [self loadConstraints];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  [self updateNavigationBarHeight];
+}
+
+- (void)viewSafeAreaInsetsDidChange {
+  [super viewSafeAreaInsetsDidChange];
+  [self updateNavigationBarHeight];
+}
+
+- (void)setSelectedViewController:(nullable UIViewController *)selectedViewController {
+  // Assert that the given VC is one of our view controllers or it is nil (we are unselecting)
+  NSAssert(
+      selectedViewController == nil || [self.viewControllers containsObject:selectedViewController],
+      @"Attempting to set BottomBarViewControllers to a view controller it does not contain");
+
+  // Early return if we are already set to the given VC
+  if (_selectedViewController == selectedViewController) {
+    return;
+  }
+
+  // Remove current VC and add new one.
+  [self removeContentViewController:self.selectedViewController];
+  [self addContentViewController:selectedViewController];
+
+  // Set the iVar and update selected index
+  _selectedViewController = selectedViewController;
+  self.selectedIndex = [self.viewControllers indexOfObject:selectedViewController];
+}
+
+- (void)setSelectedIndex:(NSUInteger)selectedIndex {
+  // If we are setting to NSNotFound deselect the items
+  if (selectedIndex == NSNotFound) {
+    [self deselectCurrentItem];
+    return;
+  }
+
+  BOOL outOfBounds = selectedIndex >= [self.viewControllers count] ||
```

Comment:
Nit: `count` is a property, please use dot notation for access (e.g., `self.viewControllers.count`)